addons/website/tests/test_website_visitor.py

```diff
@@ -513,6 +514,14 @@ def test_merge_partner_with_visitor_single(self):
         self.assertFalse(Visitor.search_count([('partner_id', '=', self.partner_admin_duplicate.id)]),
                          "The admin_duplicate visitor should've been merged (and deleted) with the admin one.")
 
+    def test_visitor_page_statistics_access(self):
+        visitor = self._get_last_visitor().with_user(self.user_demo)
+        self.assertFalse(self.user_demo.has_group("website.group_website_designer"))
+        visitor._check_field_access(visitor._fields["visitor_page_count"], "read")
+        visitor._check_field_access(visitor._fields["page_count"], "read")
+        with self.assertRaises(AccessError):
+            visitor._check_field_access(visitor._fields["page_ids"], "read")
```

Comment:
This test is checking the ACL of the field but it is not testing actually writing on the field without proper access.

We should have a test that actually does the problematic flow instead.